### DISCUSSION TITLE
How to assign adaptive lighting to a physical switch

### DISCUSSION THREAD

**Discussion by alias6023** [basnijholt/adaptive-lighting#436] [Q&A]
How can I assign adaptive lighting to a physical switch? I did the following but I need to click twice on the physical button in order to get adaptive lighting running. If I click once, the lights switch on but only 5% to 10%. As soon as I click on the physical switch again, the light changes to the correct brightness and colour. 

alias: Entrance Pico Light 1 On (Lights On)
description: ""
trigger:
  - platform: device
    device_id: f585342e9d21d015c2483b212f6a34c7
    domain: lutron_caseta
    type: press
    subtype: "on"
condition: []
action:
  - service: light.turn_on
    data: {}
    target:
      device_id: 42044ad66c9028186ce85903a76e3505
  - service: adaptive_lighting.apply
    data:
      entity_id: switch.adaptive_lighting_entrance_adaptive_lighting
  - service: adaptive_lighting.apply
    data:
      entity_id: switch.adaptive_lighting_adapt_brightness_entrance_adaptive_lighting
  - service: adaptive_lighting.apply
    data:
      entity_id: switch.adaptive_lighting_adapt_color_entrance_adaptive_lighting
mode: single

**Comment by th3w1zard1** [comment]
All you need to do is configure in your integrations, which lights adaptive-lighting will be adapting. All of those lights will then be adapted every `interval` (also set in the integrations configuration page). It's that simple.

You shouldn't have to be calling those apply services manually. [Maybe check out the readme again?](https://github.com/basnijholt/adaptive-lighting#readme)

If you really want to adapt those lights with your switch manually your script has some issues:
> leave this in:
```
service: adaptive_lighting.apply
data:
entity_id: switch.adaptive_lighting_entrance_adaptive_lighting
```

> remove these:
```
service: adaptive_lighting.apply
data:
entity_id: switch.adaptive_lighting_adapt_brightness_entrance_adaptive_lighting
service: adaptive_lighting.apply
data:
entity_id: switch.adaptive_lighting_adapt_color_entrance_adaptive_lighting
mode: single
```
